lib/graphql/batch.rb

```diff
@@ -1,6 +1,44 @@
 require "graphql"
+require "promise.rb"
+
+module GraphQL
+  module Batch
+    def self.resolve(obj)
+      obj.is_a?(Promise) ? obj : Promise.new.tap { |d| d.fulfill(obj) }
+    end
+
+    def self.all(enumerable)
+      remaining = 0
+      promise = Promise.new
+      on_fulfill = lambda do |result|
+        remaining -= 1
+        if remaining.zero?
+          promise.fulfill(enumerable.map { |obj| obj.is_a?(Promise) ? obj.value : obj })
```

Comment:
does `all` need to support a mix of promise and non-promise arguments? what is the use case for passing in non-promise args here?